### DISCUSSION TITLE
How to stop `useInifiniteScroll` firing `onLoadMore` function after all data are displayed?

### DISCUSSION THREAD

**Discussion by AloisSeckar** [vueuse/vueuse#3490] [Q&A]
I know it is called "infinite", but lets say I have 100 items - I want to display first 10, then upon scrolling down keep displaying more and more 10 until all 100 are visible. But then, how can I stop it from triyng to load more data? Ofc I can write the function with a guard, so it returns immediately, but can I prevent the callback from being called at all once it is no longer needed? Is there an option? Or should I  rather use something else than `useInfiniteScroll` for this case?

EDIT: Also bonus question - I just also noticed my callback keeps running even after leaving the page. How to properly remove it in `onUnmounted` hook?

**Comment by ilyaliao** [comment]
Maybe we can improve this, feel free to open an issue, happy to resolve it :)

**Reply by AloisSeckar** [reply]
Hi and thanks for the offer. I see this is a really old post of mine I completely forgot about.

I am not sure, if this is still an issue. At least I figured it out in my implementation:
https://github.com/AloisSeckar/InfiniTimeline/blob/main/lib/InfiniTimeline.vue#L96
Maybe it was because I didn't use `canLoadMore` function in options, or had wrong implementation of it. Anyway, it works for me.
